oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/DocumentNodeState.java

```diff
@@ -414,6 +414,11 @@ public String toString() {
      * @param revision the revision this node is created.
      */
     UpdateOp asOperation(@NotNull Revision revision) {
+
+        if (Utils.isNodeNameLong(path, store.getDocumentStore().getSizeLimit())) {
+            throw new IllegalArgumentException("Node name is too long: " + path);
```

Comment:
I'm aware the current code throws an IllegalArgumentException, but it somehow looks wrong. The path is not an argument when `asOperation(Revision)` is called. I think a `DocumentStoreException` is more appropriate.